list2/ex39/index.js

```diff
@@ -0,0 +1,26 @@
+import { generateRandomInteger } from "../helper.js";
+
+const ARRAY_NUMBERS = 100;
+
+export function compactArray() {
+  const completeArray = [];
+
+  for (let arrayIndex = 0; arrayIndex < ARRAY_NUMBERS; arrayIndex++) {
+    completeArray.push(generateRandomInteger());
+  }
+
+  const compactArray = [...completeArray];
+  compactArray.sort((a, b) => b - a);
```

Comment:
Sorting the array before compacting (compactArray.sort((a, b) => b - a)) is unnecessary and adds overhead. The problem only requires filtering values, not sorting them.